src/plugins/validation/2and3/semantic-validators/responses.js

```diff
@@ -19,7 +19,10 @@ module.exports.validate = function({ jsSpec, isOAS3 }, config) {
       each(obj, (response, responseKey) => {
         if (isOAS3) {
           each(response.content, (mediaType, mediaTypeKey) => {
-            const hasInlineSchema = mediaType.schema && !mediaType.schema.$ref;
+            const hasInlineSchema =
+              mediaType.schema &&
+              mediaTypeKey.startsWith('application/json') &&
+              !mediaType.schema.$ref;
```

Comment:
So this logic is looking good to me but I would like it implemented in a slightly different way.

Whether a response "has an inline schema" or not is independent of the content type. So including that check as part of the boolean variable changes the meaning of the variable name, which is confusing. You can just make the if statement below:

```js
if (hasInlineSchema && mediaTypeKey.startsWith('application/json') {
// ...
}
```